### PR TITLE
Various stability fixes for E2E

### PR DESCRIPTION
**Pull request by mook-as**
This PR contains various stability fixes for E2E tests; please see commit messages for details. (Reviewing each commit in order is recommended).

### CODE REVIEW

File: e2e/utils/TestUtils.ts [hunk 1]
```diff
@@ -257,6 +317,7 @@ export async function startRancherDesktop(testPath: string, options?: { tracing?
     ],
     env: {
       ...process.env,
+      ...options?.env ?? {},
       RD_LOGS_DIR: reportAsset(testPath, 'log'),
       ...options?.mock ?? true ? { RD_MOCK_BACKEND: '1' } : {},
```
Comment (ericpromislow) [inline e2e/utils/TestUtils.ts:322]: I know it's existing code, but now that more tests are calling this function, and the majority are not mocked, maybe we should make the default `false`, and then you can just do `options?.mock ? { RD_MOCK_BACKEND: 1 }: {}`

File: pkg/rancher-desktop/utils/logging.ts [hunk 2]
```diff
@@ -184,4 +205,12 @@ export function clearLoggingDirectory(): void {
   }
 }
 
+export function reopenLogs() {
+  for (const log of logs.values()) {
+    log['reopen']('a');
+    // Trigger making the stream
+    ((_: any) => {})(log.fdStream);
```
Comment (ericpromislow) [inline pkg/rancher-desktop/utils/logging.ts:212]: This incantation needs some explanation. Looks like we're calling a function, but which function are we calling? And what does the `_` in `(_: any)` mean anyway?

File: pkg/rancher-desktop/utils/logging.ts [hunk 3]
```diff
@@ -38,10 +38,40 @@ export function setLogLevel(level: logLevel): void {
 export class Log {
   constructor(topic: string, directory = paths.logs) {
     this.path = path.join(directory, `${ topic }.log`);
-    this.stream = fs.createWriteStream(this.path, { flags: 'w', mode: 0o600 });
+    this.reopen();
+    // The following lines only exist because TypeScript can't reason about
```
Comment (ericpromislow) [inline pkg/rancher-desktop/utils/logging.ts:42]: Could you add an explanation of why these new functions are all needed?

File: e2e/utils/TestUtils.ts (hunk 1 above)
Comment (mook-as) [reply e2e/utils/TestUtils.ts:322]: The default is mocked to encourage use of the mock backend (which is much faster); the extra effort required to use the non-mocked backend is intentional.

**Review by ericpromislow — CHANGES_REQUESTED** [review]
The code is fine (good point on mocking), and there are no regressions. I need some comments at the two indicated points to understand that part of this change.

File: e2e/containers.e2e.spec.ts [hunk 4]
```diff
@@ -18,28 +18,16 @@ let page: Page;
  * */
 test.describe.serial('Container Engine', () => {
   let electronApp: ElectronApplication;
-  let context: BrowserContext;
+
+  test.beforeAll(async() => {
```
[the remark this answers is not in the document]
Comment (adamkpickering) [reply e2e/containers.e2e.spec.ts]: Ok, we can leave this :+1:

File: e2e/utils/TestUtils.ts [hunk 5]
```diff
@@ -157,6 +157,31 @@ export async function teardown(app: ElectronApplication, filename: string) {
     if (proc.kill('SIGTERM') || proc.kill('SIGKILL')) {
       console.log(`Manually stopped process ${ pid }`);
     }
+    // Try to do platform-specific killing based on process groups
+    if (process.platform === 'darwin' || process.platform === 'linux') {
+      for (const signal of ['TERM', 'TERM', 'TERM', 'KILL']) {
+        let pids = '';
+
+        try {
+          const args = ['-o', 'pid=', process.platform === 'darwin' ? '-g' : '--sid', `${ pid }`];
```
[the remark this answers is not in the document]
Comment (adamkpickering) [reply e2e/utils/TestUtils.ts:172]: Whoops, missed that.

After learning about this a bit more, I'm confident that this is a good way of doing this.

**Review by adamkpickering — APPROVED** [review]

**Comment by jandubois** [comment]
@ericpromislow Please verify if your request for changes has been satisfied!

File: pkg/rancher-desktop/utils/logging.ts (hunk 3 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/utils/logging.ts:42]: ok

**Review by ericpromislow — APPROVED** [review]
It's fine now. Thanks for the clarifications.

**Pull request merged**